### PR TITLE
Parse and Expose the custom_permissions returned in ID Url response

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: libs/SalesforceOAuth/SalesforceOAuth/Classes/SFOAuthCoordinator.m [hunk 1]
```diff
@@ -281,21 +282,13 @@ - (void)beginUserAgentFlow {
     } else {
         [approvalUrl appendFormat:@"&%@=%@", kSFOAuthResponseType, kSFOAuthResponseTypeToken];        
     }
-
```
Comment (wmathurin) [inline libs/SalesforceOAuth/SalesforceOAuth/Classes/SFOAuthCoordinator.m]: Refactored to be more readable

File: libs/SalesforceOAuth/SalesforceOAuth/Classes/SFOAuthCoordinator.m [hunk 2]
```diff
@@ -281,21 +282,13 @@ - (void)beginUserAgentFlow {
...
-        }
-        
-        NSString *finalScopeStr = [scopeStr stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
-        [approvalUrl appendString:finalScopeStr];
-    }
+    
+    // Adding refresh_token and custom_permissions scopes
+    NSMutableSet* scopes = [NSMutableSet setWithSet:self.scopes];
+    [scopes addObject:kSFOAuthRefreshToken];
+    [scopes addObject:kSFOAuthCustomPermissions];
```
Comment (wmathurin) [inline libs/SalesforceOAuth/SalesforceOAuth/Classes/SFOAuthCoordinator.m]: We need the custom_permissions scope to get the custom permissions back in the id response

**Comment by bhariharan** [comment]
:+1:

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityData.m [hunk 3]
```diff
@@ -322,6 +323,11 @@ - (NSDictionary *)customAttributes
     return (self.dictRepresentation)[kSFIdentityCustomAttributesKey];
 }
 
+- (NSDictionary *)customPermissions
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityData.m]: This should have a corresponding property definition in `SFIdentityData.h`.

**Comment by khawkins** [comment]
Otherwise, looks good to me.

**Comment by khawkins** [comment]
Thinking about it, I'm not sure we want to de facto ask for this permission.  This isn't something that everyone is going to use, and making it a required field essentially creates a lot of extra documentation and explanation to the majority of our users who wouldn't otherwise need to know anything about it.

Should this not simply be a documentation task for the few people who will use this functionality?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityData.m (hunk 3 above)
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityData.m]: Good catch, done here: https://github.com/wmathurin/SalesforceMobileSDK-iOS/commit/12a40107d38888816e64373a8fcb226e97187721

**Comment by wmathurin** [comment]
@khawkins it probably shouldn't be required, I'll remove it from the code, but we should document that apps that want to have custom permissions need to have the custom permissions in their connected app scopes and in their bootconfig scopes.

**Comment by wmathurin** [comment]
custom_permissions scope not longer added in, see https://github.com/wmathurin/SalesforceMobileSDK-iOS/commit/fe4bd1a044ffd3f802fdea29cbfe542d93b37b9d

**Comment by khawkins** [comment]
Looks good.

**Comment by khawkins** [comment]
@rwhitley This will be a candidate for documentation around Connected App settings.

**Pull request merged**